readme.md

```diff
@@ -145,6 +146,34 @@ add2(10) // 12
 
 ```
 
+##Closure
+
+A very simplistic definition - A closure is a way of accessing a variable outside its scope.
+Formally, a closure is a technique for implementing lexically scopped named binding. It is a way of storing a function with an environment. 
+
+A closure is a scope which captures local variables of a function for access even after the execution has moved out of the block in which it is defined.
+ie. they allow referencing a scope after the block in which the variables were declared has finished executing. 
+
+
+```js
+function getTicker () {
+  var tick = 0;
+  function ticker () {
+   tick = tick + 1;
+   return tick;
+   }
+  return ticker;
+ }
+ var tickTock = getTicker();
+ tickTock(); //returns 1
+ tickTock(); //returns 2
```

Comment:
I know. This  did come to my mind. But thinking about it, aren't closures kind of a side effect? Function's scope is over and yet incrementing a value? I couldn't think of a pure example :(
The best I can think of is to have an alert() instead of tick+=1, sounds like a sideeffect too